From 82b9fab203cfa800bdfda95b8611b20f80f89c29 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 02 Nov 2020 18:58:23 +0200
Subject: [PATCH] [SW-12754] gaudi: security: fix pcie rrs val mask

The commit:
792e513 ("[SW-12754] gaudi: security: rename pcie rrs func")
while did a correct renaming, but did not accound for the semantic
changes and thus the offsets were masked with HL_CFG_BASE which is
wrong if protecting HBW regions.
Fix this by applying a correct mask - cut the upper 32 bits.
And also supplying the correct offsets for protection around GIC.

Change-Id: I1f77745fd7dffb936b812ef150df0cac70049882
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index c8ea1e6..70eea09 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1048,8 +1048,8 @@
 
 static void preboot_secure_pcie_rr(int rr, u64 min64, u64 max64)
 {
-	u32 min = min64 & ~HL_CFG_BASE;
-	u32 max = max64 & ~HL_CFG_BASE;
+	u32 min = (u32)min64;
+	u32 max = (u32)max64;
 	u32 rr_base = rr * sizeof(u32), rr_dest;
 
 	/* Set WP RR */
@@ -1117,12 +1117,12 @@
 
 	/* HBM_CFG up to GIC_L: RR_2 */
 	min = mmHBM0_BASE - 1;
-	max = mmGIC_DISTRIBUTOR__5_GICD_SETSPI_NSR;
+	max = HL_CFG_BASE + mmGIC_DISTRIBUTOR__5_GICD_SETSPI_NSR;
 	preboot_secure_lbw_rr(2, min, max);
 	preboot_secure_pcie_rr(2, min, max);
 
 	/* GIC_H up to PCIE_WRAP: RR_3 */
-	min = mmGIC_DISTRIBUTOR__5_GICD_SETSPI_NSR + sizeof(u32) - 1;
+	min = max + sizeof(u32) - 1;
 	max = mmPCIE_WRAP_BASE;
 	preboot_secure_lbw_rr(3, min, max);
 	preboot_secure_pcie_rr(3, min, max);
